### DISCUSSION TITLE
Support ConfiguredGraphFactory configuration in JanusGraph docker

### DISCUSSION THREAD

**Discussion by eanton86** [JanusGraph/janusgraph#2362] [General]
Hello, JanusGraph maintainers! 
I'm working on a JanusGraph setup managed by ConfiguredGraphFactory ([documentation](https://docs.janusgraph.org/basics/configured-graph-factory/)). I'm using an usual [JanusGraph docker image](https://github.com/JanusGraph/janusgraph-docker) and found that there is no straight way to make it work using current options (mounted configuration aside).  
Basically, we need two configuration files to set up: gremlin-server.yml and janusgraph.properties. There are no problems to customize janusgraph.properties. We can either use JANUS_PROPS_TEMPLATE variable to pick up some configuration template or provide any janusgraph.* variables, or use both. 
The problem is with gremlin-server.yml. It requires many modifications, but we don't have the same template variable for it. Using of gremlinserver.* variables doesn't help much, because of how those properties are replaced in gremlin-server.yml, e.g. with current scripts we can not replace this line 
```
graph: /etc/opt/janusgraph/janusgraph.properties
```
to 
```
ConfigurationManagementGraph: /etc/opt/janusgraph/janusgraph.properties
```
We can make it work better by doing the following: 
1. The main thing - let's apply JANUS_PROPS_TEMPLATE for gremlin-server.yml as well. There are gremlin templates in distribution already that correspond to options of JANUS_PROPS_TEMPLATE  (e.g. gremlin-server-cql-es.yml). Only changes in docker-endpoint.sh required. 
2. So we can pass in JANUS_PROPS_TEMPLATE, let's say, : cql-configurationgraph to apply gremlin-server-cql-configurationgraph.yml along with janusgraph-cql-configurationgraph.properties. These files are already there, but have different names and locations:

[conf\gremlin-server\gremlin-server-configuration.yaml](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/static/conf/gremlin-server/gremlin-server-configuration.yaml)
[conf\janusgraph-cql-configurationgraph.properties](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/cfilter/conf/janusgraph-cql-configurationgraph.properties)

So need to move to common folder (gremlin-server) and rename to follow the pattern. 
3. Optionally add docker-compose-cql-configurationgraph.yml to make it ready as a whole, so people won't follow the hard way I did. 

If my proposal is eligible I can come up with a pull request. What do people think?

**Comment by farodin91** [comment]
There is already a pr request. I have completely forgot the pr https://github.com/JanusGraph/janusgraph-docker/pull/55.

I will try get the pr out as quick as possible.

**Reply by eanton86** [reply]
Thanks for pointing out this pr. I will try the new configuration options.
